packages/peregrine/lib/talons/Checkout/useForm.js

```diff
@@ -0,0 +1,13 @@
+import { useQuery } from '@apollo/react-hooks';
+
+export const useForm = props => {
+    const { countriesQuery } = props;
+
+    const { data } = useQuery(countriesQuery);
+
+    const { countries } = data || {};
```

Comment:
Now that I read this I realize in the past we would make requests for country data in the `beginCheckout` async action which was `await`ed in the initial opening of the submission form. This means that we don't render the form until we already have the data.

Now, we could possibly render with an empty object for `countries` which may be incorrect. I think this means you should add a loading spinner/state and use that when countries is empty (or null/undefined). In this case it's probably safe to just return `data` since it'll be undefined until then and we can block render until we get an error or it's finished loading.

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/1993/commits/4a253fee5ae2a149102f83edf2c7c5767d3fc1db.

**Loading:**

![loading_checkout](https://user-images.githubusercontent.com/13182778/69565976-a90fcb00-0f7b-11ea-9dfd-76466837d000.gif)


**Error:**

![error_checkout](https://user-images.githubusercontent.com/13182778/69566328-71555300-0f7c-11ea-8959-dd994ae2087f.gif)

Let me know what you think of the implementation for sure.